File: apps/worker/src/instrument.ts
import * as Sentry from '@sentry/node';

const dsn = process.env.SENTRY_DSN;

if (dsn) {
  const release = process.env.SENTRY_RELEASE;
  const environment = process.env.NODE_ENV ?? 'development';
  const tracesSampleRate = environment === 'production' ? 0.1 : 1.0;

  const initOptions: Sentry.NodeOptions = {
    dsn,
    environment,
    tracesSampleRate,
    sendDefaultPii: false,
    beforeSend(event) {
      if (event.user) {
        delete event.user.email;
        delete event.user.ip_address;
      }
      return event;
    },
  };

  if (release) {
    initOptions.release = release;
  }

  Sentry.init(initOptions);
}

export { Sentry };

File: apps/inference/app/instrument.py
import os

import sentry_sdk
from sentry_sdk.integrations.fastapi import FastApiIntegration


def init_sentry() -> None:
    dsn = os.getenv("SENTRY_DSN")
    if not dsn:
        return
    sentry_sdk.init(
        dsn=dsn,
        release=os.getenv("SENTRY_RELEASE"),
        environment=os.getenv("NODE_ENV", "development"),
        traces_sample_rate=0.1 if os.getenv("NODE_ENV") == "production" else 1.0,
        send_default_pii=False,
        integrations=[FastApiIntegration()],
    )
